packages/nextcloud_test/lib/src/test_client.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:nextcloud/core.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:nextcloud_test/src/docker_container.dart';
+import 'package:process_run/process_run.dart';
+import 'package:universal_io/io.dart';
+
+/// An extension for creating [NextcloudClient]s based on [DockerContainer]s.
+extension TestNextcloudClient on NextcloudClient {
+  /// Creates a new [NextcloudClient] for a given [container] and [username].
+  ///
+  /// It is expected that the password of the user matches the its [username].
+  /// This is the case for the available test docker containers.
+  static Future<NextcloudClient> create(
+    final DockerContainer container, {
+    final String? username = 'user1',
```

Comment:
In what situation does it make sense to have a null username?
Our tests currently don't make use of it.

Or do we need it to test public apis?

Reply:
Yes that will be the use-case I think. We can also make it non-nullable for now until we need it.